jme3-effects/src/main/java/com/jme3/water/WaterFilter.java

```diff
@@ -120,7 +122,9 @@
     private Vector3f center;
     private float radius;
     private AreaShape shapeType = AreaShape.Circular;
-    
+
+    private boolean needSaveReflectionScene;
```

Comment:
Yes I know, my point, is why do we need the flag? why not save it in any case.
Also... saving the scene here may be a problem for j3f in the SDK. Technically J3o works with object references, so an object data is effectively saved once even if it's referenced multiple times. But in case of j3f only the filter stack is saved, doing this you might save the whole scene into a j3f... Of course it's not a problem if you save all in one j3o...but the sdk doesn't do that.